src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -487,6 +488,75 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - This method is called when the user clicks the "export message history" button
+    //   in the query palette
+    // Arguments:
+    // - text - the text to export
+    // Return Value:
+    // - <none>
+    void TerminalPage::_OnExportChatHistoryRequested(const IInspectable& /*sender*/, const winrt::hstring& text)
+    {
+        _SaveFileHelper(text, L"", L"");
+    }
+
+    fire_and_forget TerminalPage::_SaveFileHelper(const winrt::hstring& text, const winrt::hstring& filepath, const winrt::hstring& filename)
+    {
+        // This will be used to set up the file picker "filter", to select .txt
+        // files by default.
+        static constexpr COMDLG_FILTERSPEC supportedFileTypes[] = {
+            { L"Text Files (*.txt)", L"*.txt" },
+            { L"All Files (*.*)", L"*.*" }
+        };
+        // An arbitrary GUID to associate with all instances of this
+        // dialog, so they all re-open in the same path as they were
+        // open before:
+        static constexpr winrt::guid clientGuidExportFile{ 0xF6AF20BB, 0x0800, 0x48E6, { 0xB0, 0x17, 0xA1, 0x4C, 0xD8, 0x73, 0xDD, 0x58 } };
+
+        try
+        {
+            auto path = filepath;
+
+            if (path.empty())
+            {
+                // GH#11356 - we can't use the UWP apis for writing the file,
+                // because they don't work elevated (shocker) So just use the
+                // shell32 file picker manually.
+                std::wstring cleanedFilename{ til::clean_filename(filename.c_str()) };
```

Comment:
This gained an extra `.c_str()`. You can avoid this by casting the `hstring` to a `wstring_view`. It's even easier to avoid this if you change the `filename` parameter type from `hstring` to `wstring_view`.

Reply:
Oh the `wstring` constructor is explicit, so you have to do `clean_filename(std::wstring{ filename })` anyway. What a weird language...